utils/src/check/clobber-functions.c

```diff
@@ -140,6 +140,53 @@ static struct clobber_function clobber_pb_qconf_wrong_version = {
 	&do_clobber_pb_qconf_wrong_version,
 };
 
+static int pick_random_qslot(void)
+{
+	int j = 0;
+	int filled_slots[SCOUTFS_QUORUM_MAX_SLOTS];
+	uint16_t family;
+	srandom(time(NULL));
+
+	for (int i = 0; i < SCOUTFS_QUORUM_MAX_SLOTS; i++) {
+		family = le16_to_cpu(global_super->qconf.slots[i].addr.v4.family);
+		if (family == SCOUTFS_AF_NONE)
+			continue;
+
+		filled_slots[j] = i;
+		j++;
+	}
+	return filled_slots[(random() & (j-1))];
```

Comment:
okay, but, use `%` instead here. While the `&` will work it doesn't create a truly random contstraint and will quite likely never pick some of the slots. For instance, if there are 5 quorum slots, the result is `random() & 4` which isn't ever going to result in `3`. The `%` operator however will do what you want.